app.js

```diff
@@ -1,90 +1,156 @@
                 }
 
+                const colsName = Object.keys(data[0]);
+                let createTableQuery = `CREATE TABLE ${tableName} (`;
+                let connection = mysql.createConnection(DB_CONFIG);
                 let completedQueryCount = 0;
-                data.forEach((row, index, array) => {
-                    let rowToInsert = [];
-                    for (let col in row) {
-                        rowToInsert.push(row[col]);
-                    }
 
-                    connection.query(`INSERT INTO ${tableName} VALUES (${rowToInsert.map(x => {
-                        x = x.replace(/"/g, '\\"');
-                        return '"' + x + '"';
-                    }).join()})`, error => {
-                        // Close the connection once all queries are complete
-                        if (++completedQueryCount === array.length) {
-                            connection.end();
+                if (insertingCase !== "--append") {
+                    if (colsName.join().indexOf(" ") > -1 || colsName.join().indexOf("'") > -1 || colsName.join().indexOf('"') > -1) {
+                        for (let column of colsName) {
+
+                            if (column.indexOf(" ") > -1 || column.indexOf("'") > -1 || column.indexOf('"') > -1) {
+                                createTableQuery += '`' + column + '`' + " TEXT, ";
+                            } else {
+                                createTableQuery += column + " TEXT, ";
+                            }
                         }
+                        createTableQuery = createTableQuery.slice(0, -2) + ")";
+                    } else {
+                        createTableQuery += colsName.join(" TEXT, ") + " TEXT)";
+                    }
+                }
+                if (insertingCase === "--overwrite") {
 
+                    connection.query(`DROP TABLE IF EXISTS ${tableName.replace(/'/g, "\\'")}`, error => {
                         if (error) {
                             console.error(error);
                             return;
                         }
+                        CreateTable(createTableQuery, data, tableName, insertRows, completedQueryCount, connection);
                     });
-                });
+
+                } else if (insertingCase === "--append") {
+
+                    insertRows(tableName, data, completedQueryCount, connection);
+
+                } else {
+
+                    CreateTable(createTableQuery, data, tableName, insertRows, completedQueryCount, connection);
+
+                }
             });
         });
     });
-});
+}
+
+
+let args = process.argv;
+filesToInsert = [];
+for (let i = 2; i < args.length; i++) {
+    if (args[i] === "--files") {
+        isUserSpecifiedFilesToInsert = true;
+    }
+    if (args[i] === '--append' || args[i] === '--overwrite') {
+        insertingCase = args[i];
+    }
+    if (args[i].indexOf('.csv') > -1 && isUserSpecifiedFilesToInsert) {
```

Comment:
If a user does the following command:
`node app.js --files test.txt`
And test.txt is a valid CSV file, then the file will not be inserted and the user will see an error saying `No CSV files specified` instead
I think you should fix this and add a test case for it